Lib/typing.py

```diff
@@ -818,6 +869,117 @@ def __init__(self, name, *constraints, bound=None,
             self.__module__ = def_mod
 
 
+class TypeVarTuple(_Final, _Immutable, _root=True):
```

Comment:
I meant there seems to be no reason why `TypeVarTuple` can't inherit from `_BoundVarianceMixin` (previously `TypeVarLike`). Someone can use `isinstance(x, TypeVarLike)` instead of hardcoding `isinstance(x, (TypeVar, ParamSpec))` and thus missing out `TypeVarTuple`.

The downside is that `TypeVarTuple("Ts", bound=...)` or `covariant=True` will be valid at runtime. That's fine. We want the runtime to be as lenient as possible so that we can specify the behavior of `bound` in the future. The counter-argument is that we might want a different name such as `item_bound` for TypeVarTuple.

I'm ok with either.

Reply:
Not being able to do `isinstance(x, TypeVarLike)` is a definite downside, though with the current implementation where `Unpack` is implemented using `_SpecialForm`, I think we won't be able to check using `isinstance` anyway (at least not for unpacked `TypeVarTuple`, which I think is what we'd care about). I have added a helper function `_is_typevar_like`, though, for exactly this purpose of making sure we don't miss out `TypeVarTuple`; I hope that should work well enough, and we can make it public in the future if there's demand for it.

As for whether `bound` and `covariant` should be accepted at runtime: I remain hesitant about this degree of leniency, and I think the point you raise about us maybe wanting to use a different name for `bound` is actually a pretty important one. So in the end, I'm not too worried about this either.